cern_access/indico_cern_access/util.py

```diff
@@ -0,0 +1,294 @@
+
+from indico.core.notifications import make_email, send_email
+from indico.modules.events.registration.models.forms import RegistrationForm
+from indico.modules.events.registration.models.registrations import Registration
+from indico.modules.events.registration.util import get_ticket_attachments
+from indico.util.string import remove_accents, unicode_to_ascii
+from indico.web.flask.templating import get_template_module
+
+from indico_cern_access import _
+from indico_cern_access.models.access_request_regforms import CERNAccessRequestRegForm
+from indico_cern_access.models.access_requests import CERNAccessRequest, CERNAccessRequestState
+
+
+def get_requested_forms(event):
+    """Returns list of registration forms with requested access to CERN"""
+    return (RegistrationForm.query.with_parent(event)
+            .join(CERNAccessRequestRegForm)
+            .filter(CERNAccessRequestRegForm.is_active)
+            .all())
+
+
+def get_requested_registrations(event, regform=None):
+    """By default returns a list of requested registrations of an event
+
+    :param regform: if specified, returns only registrations with that registration form
+    """
+    query = (Registration.query.with_parent(event)
+             .join(CERNAccessRequest)
+             .filter(CERNAccessRequest.is_active))
+    if regform:
+        query = query.filter(Registration.registration_form_id == regform.id)
+    return query.all()
+
+
+def send_adams_post_request(event, registrations, update=False):
+    """ Sends POST request to ADaMS API
+
+    :param update: if True, send request updating already stored data
+    """
+    from indico_cern_access.plugin import CERNAccessPlugin
+    data = {registration.id: build_access_request_data(registration, event, update=update)
+            for registration in registrations}
+    headers = {'content-type': 'application/json'}
+    auth = (CERNAccessPlugin.settings.get('login'), CERNAccessPlugin.settings.get('password'))
+    json_data = json.dumps(data.values())
+    url = CERNAccessPlugin.settings.get('adams_url')
+    try:
+        r = requests.post(url, data=json_data, headers=headers, auth=auth)
+        r.raise_for_status()
+    except requests.exceptions.RequestException:
+        CERNAccessPlugin.logger.exception('Request to ADAMS failed (%r)', json_data)
+        raise AdamsError(_('Sending request to ADAMS failed'))
+    return (CERNAccessRequestState.accepted, data)
+
+
+def send_adams_delete_request(registrations):
+    """Sends DELETE request to ADaMS API"""
+    from indico_cern_access.plugin import CERNAccessPlugin
+
+    data = [generate_access_id(registration.id) for registration in registrations]
+    headers = {'Content-Type': 'application/json'}
+    auth = (CERNAccessPlugin.settings.get('login'), CERNAccessPlugin.settings.get('password'))
+    data = json.dumps(data)
+    url = CERNAccessPlugin.settings.get('adams_url')
+    try:
+        r = requests.delete(url, data=data, headers=headers, auth=auth)
```

Comment:
I would do the jsonification here. Like this `data` in the exception below is still the python structure (which is nicer when looking at the exception log or the locals in sentry)